contracts/core/Vault.sol

```diff
@@ -186,4 +210,24 @@ contract Vault is
         // Return owners token balance
         return balances[_token][_owner];
     }
+
+    /* ============ Internal Functions ============ */
+
+    function transferBalanceInternal(
```

Comment:
The naming scheme in the vault file is a little weird. A few suggestions:
- Does it make sense for it to be caleld `incrementTokenOwner` and `decrementTokenOwner`? Everything else is suffixed with `Balance` now which I like since it's explicit about the property being mutated
- CoreAccounting has an `internalTransfer`. Let's rename this to something like `transferVaultBalanceInternal`, keeping to the `internal` suffix and being explicit that it's for vault only